server-backend/src/LiquidService.ts

```diff
@@ -0,0 +1,112 @@
+        };
+        this.configurationDetails = {
+            rpcUrl: this.rpcUrl,
+            rpcAuth: this.rpcAuth,
+            xpub: this.xpub,
+        };
+    }
+    
+    async onApplicationBootstrap(): Promise<void> {
+        this.logger.debug('Starting to initialize LiquidService xpub');
+        try {
+            await this.nbxplorer.track(this.xpub, 'lbtc');
+            this.logger.debug('LiquidService xpub initialized successfully');
+        } catch (error: unknown) {
+            const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+            this.logger.error(`Failed to initialize Liquid xpub: ${errorMessage}`);
+        }
+    }
+
+    async callRPC(method: string, params: any[] = [], walletName: string = ''): Promise<any> {
+        const url = walletName ? `${this.rpcUrl}/wallet/${walletName}` : this.rpcUrl;
+        try {
+            const response = await axios.post(url, {
+                jsonrpc: '1.0',
+                id: '40swap',
+                method,
+                params,
+            }, {
+                auth: this.rpcAuth,
+            });
+            return response.data.result;
+        } catch (error) {
+            this.logger.error(`Error calling Elements RPC ${method}: ${(error as any).message}`);
+            throw error;
+        }
+    }
+
+    async getUtxos(): Promise<NBXplorerUtxosResponse | void> {
+        if (!this.xpub) {
+            this.logger.error('Attempting to get UTXOs with empty xpub');
+            throw new Error('Xpub is not initialized');
+        }
+        const utxoResponse = await this.nbxplorer.getUTXOs(this.xpub, 'lbtc');
+        if (!utxoResponse) {
+            throw new Error('No UTXOs returned from NBXplorer');
+        }
+        return utxoResponse;
+    }
+
+    async getConfirmedUtxos(): Promise<NBXplorerUtxo[]> {
+        const utxoResponse = await this.getUtxos();
+        return utxoResponse?.confirmed.utxOs ?? [];
+    }
+
+    async getConfirmedUtxosAndInputValueForAmount(amount: number): Promise<{ 
+        utxos: NBXplorerUtxo[], 
+        totalInputValue: number,
+    }> {
+        let totalInputValue = 0;
+        const confirmedUtxos = await this.getConfirmedUtxos();
+        if (confirmedUtxos.length === 0) {
+            throw new Error('No confirmed UTXOs found');
+        }
+        const selectedUtxos = [];
+        for (const utxo of confirmedUtxos) {
+            selectedUtxos.push(utxo);
+            totalInputValue += Number(utxo.value);
+            if (totalInputValue >= amount) {
+                break;
+            }
+        }
+        if (totalInputValue < amount) {
+            throw new Error(`Insufficient funds, required ${amount} but only ${totalInputValue} available`);
+        }
+        return { utxos: selectedUtxos, totalInputValue };
+    }
+  
+    async signPsbt(psbtBase64: string): Promise<string> {
+        return this.callRPC('walletprocesspsbt', [psbtBase64]);
```

Comment:
we can't be sure that the rpc call is returning a string, we should validate it with zod.
```suggestion
        return z.string().parse(this.callRPC('walletprocesspsbt', [psbtBase64]));
```

Reply:
Function removed for now